### DISCUSSION TITLE
Behaviour of `removeparam` when using end of string identifier (`$`)

### DISCUSSION THREAD

**Discussion by partingscientist** [uBlockOrigin/uBlock-issues#2685] [Q&A] [answered]
Given the following link: `https://example.com/?foo`, 
the query parameter `foo` cannot be removed using `||example.com^$removeparam=/^foo$/`. 

I first thought that opening `https://example.com/?foo` perhaps means sending a GET request to `https://example.com/?foo=`, which explains `||example.com^$removeparam=/^foo=$/` being able to remove the parameter. However, a similar filter form has been referenced before [[1]](https://github.com/uBlockOrigin/uBlock-issues/wiki/Static-filter-syntax#_-aka-noop) in the documentation.

Is this behaviour intended? Did I miss something? Thank you.

**Comment by gorhill** [comment]
Documentation needs to be fixed.

**Comment by ghost** [comment, marked as answer]
>perhaps means sending a GET request to `https://example.com/?foo=`, which explains...

This is because:

>When using a literal regular expression, it gets tested against each query parameter name-value pair assembled into a single string as `name=value`.

uBO is doing the "assembling" internally, that's why there is the equation sign. https://github.com/gorhill/uBlock/blob/257cfad90d515f4c6fb03f8f2ec3fbf3f3fd6487/src/js/static-net-filtering.js#L5425
